libsplinter/src/admin/rest_api/actix/circuits.rs

```diff
@@ -164,36 +164,39 @@ fn query_list_circuits<T: CircuitStore + 'static>(
 mod tests {
     use super::*;
 
-    use crate::actix_web::{
-        http::{header, StatusCode},
-        test, web, App,
-    };
+    use reqwest::{blocking::Client, StatusCode, Url};
+
     use crate::circuit::{
         directory::CircuitDirectory, AuthorizationType, Circuit, DurabilityType, PersistenceType,
         Roster, RouteType, ServiceDefinition, SplinterState,
     };
-    use crate::rest_api::paging::Paging;
+    use crate::rest_api::{paging::Paging, RestApiBuilder};
     use crate::storage::get_storage;
 
     #[test]
     /// Tests a GET /admin/circuits request with no filters returns the expected circuits.
+    ///
+    /// This test binds to `127.0.0.1:8092`; this URL should not be bound to by other tests.
     fn test_list_circuits_ok() {
-        let splinter_state = filled_splinter_state();
-
-        let mut app = test::init_service(
-            App::new().data(splinter_state.clone()).service(
-                web::resource("/admin/circuits")
-                    .route(web::get().to_async(list_circuits::<SplinterState>)),
-            ),
-        );
+        let bind_url = "127.0.0.1:8092";
 
-        let req = test::TestRequest::get().uri("/admin/circuits").to_request();
+        let (_shutdown_handle, _join_handle) = RestApiBuilder::new()
+            .with_bind(bind_url)
+            .add_resource(make_list_circuits_resource(filled_splinter_state()))
+            .build()
+            .expect("Failed to build REST API")
+            .run()
+            .expect("Failed to run REST API");
 
-        let resp = test::call_service(&mut app, req);
+        let url = Url::parse(&format!("http://{}/admin/circuits", bind_url))
+            .expect("Failed to parse URL");
+        let req = Client::new()
+            .get(url)
+            .header("SplinterProtocolVersion", protocol::ADMIN_PROTOCOL_VERSION);
```

Comment:
Interested to hear peoples' thoughts on if we should also add tests for other protocol versions; may not be necessary right now since everything is 1, but when these versions get updated tests should probably be added for old versions.

Reply:
Agree, but is there anything to do now?

Reply:
Don't think so; we just have to be sure (as reviewers) going forward that we make sure these tests get added with new versions.